kubernetes/client/doc.go

```diff
@@ -0,0 +1,91 @@
+// Copyright © 2018 Genome Research Limited
+// Author: Theo Barber-Bany <tb15@sanger.ac.uk>.
+//
+//  This file is part of wr.
+//
+//  wr is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  wr is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with wr. If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+/*
+
+Package client provides functions to interact with a kubernetes cluster, used to
+create resources so that you can spawn runners, then delete those
+resources when you're done. Everything is centred around a Kubernetesp struct.
+Calling Authenticate() will allow AttachCmd and ExecCmd to work without needing
+to Initialize()
+
+This package uses client-go: https://github.com/kubernetes/client-go
+
+Authenticate will try 3 ways to authenticate against a cluster, initially it will try
+and read in the masterUrl and kubeconfigPath from the users home directory, if that fails then
+it will attempt to check the env variable KUBECONFIG for a path to alternate configuration.
+Finally it will fall back to in cluster config which uses the service account kubernetes gives
+to pods, expecting to be run in a kubernetes environment. It also returns and stores a clientset,
+clusterConfig and REST Client in p.clientset, p.clusterConfig, p.RESTClient. The reason it's done
+like this is because it allows a lighter use of the client library (e.g tearing down, exec various
+commands) without needing to call Initialize()
+
+Initialize() takes the passed clientset and optionally a namespace and creates various clients
```

Comment:
What's a clientset? What's a namespace? I haven't gone through it, but if someone's only experience of k8s is completing https://kubernetes.io/docs/tutorials/kubernetes-basics/create-cluster/cluster-intro/, can they come to this documentation and understand what's going on, and deploy something using this library to their new cluster?